components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/validationhandler/impl/RoleBasedScopeValidationHandler.java

```diff
@@ -100,6 +102,10 @@ public List<String> validateScopes(List<String> requestedScopes, List<String> ap
                     .collect(Collectors.toList());
             return requestedScopes.stream().filter(filteredScopes::contains).collect(Collectors.toList());
         } catch (IdentityOAuth2Exception | IdentityRoleManagementException e) {
+            if (e instanceof IdentityOAuth2ClientException) {
```

Comment:
The exception handling can be done properly as suggested in this PR https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2513

Reply:
This is now updated to match the PR https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2513